### ISSUE TITLE
Trigger a notification when metadata is submitted via the submission portal

### ISSUE THREAD

**Issue by mslarae13** [microbiomedata/issues#415]
**Deliverable this task is associated with**

_See Deliverables tab here: https://docs.google.com/spreadsheets/d/1z_b6WbuTk4pI0Q-Z-rfCgC-8R3m3F2_JDevYuK8CjYE/edit?usp=sharing_

- 5

**RACI**

_Tag people in their roles_
- Responsible: 
- Accountable: @pkalita-lbl 
- Consulted:
- Informed:

**Describe the the task?**
- [ ] ~Create a notification to @mslarae13 & ??? when a user submits metadata / status changes to pending~
- [ ] ~Identify who should be notified about metadata submission~
- [ ] ~When metadata submission is for submission to DOE UF, trigger notification to EMSL MPOC &/or JGI PM~
- [ ] Automatic issue creation when metadata is submitted.

**Criteria for completion**
- An email notification is triggered for submissions

**Estimate people time**
- ?

**Completion Date (Goal)**
- ~11/17~
- Rescheduled, 03/15 (TBC)

**Target Sprint Start & End Dates**
- Start: 10/23
- End: 11/17

**Tag Blocker/Contingent upon issues**
- [Tagg issues]

**Comment by mslarae13** [comment]
Github issue creation prioritized, all other triggered notifications cancelled for this squad

**Comment by mslarae13** [comment]
This is hard. And is not currently prioritized.

**Comment by mslarae13** [comment]
Need a github account to create the issues. Need to discuss with leadership 
@ssarrafan  can you add to Friday's agenda

**Comment by ssarrafan** [comment]
> Need a github account to create the issues. Need to discuss with leadership @ssarrafan can you add to Friday's agenda

This is on the agenda for this Friday's meeting.  If you need something sooner let me know.

**Comment by JamesTessmer** [comment]
> > Need a github account to create the issues. Need to discuss with leadership @ssarrafan can you add to Friday's agenda
> 
> This is on the agenda for this Friday's meeting. If you need something sooner let me know.

There’s still some other work to do that isnt blocked by this, so I don’t anticipate needing this sooner tham Friday.

**Comment by naglepuff** [comment]
Info on organizational accounts if this is the route we pursue: https://docs.github.com/en/get-started/learning-about-github/types-of-github-accounts#organization-accounts

**Comment by mslarae13** [comment]
Discussed at leadership meeting.
Deprioritized the CRM, too much work & low priority
Was still expecting that to make a github ticketing system. 

@JamesTessmer confirmed / approved. Good to keep moving ahead. 
Might eventually pair that with another system but regardless there would still be a github ticket.

Automated process under organizational account-
- yes, but we need to look into that. Currently don't have an organizational account
- You can generate tokens for specific use cases. 
- Keep progressing just using your account, and we'll start the process of getting an organizational account and switch over.

**Comment by JamesTessmer** [comment]
Here is the PR for this issue

https://github.com/microbiomedata/nmdc-server/pull/1208

**Comment by mslarae13** [comment]
@JamesTessmer are you working on @pkalita-lbl 's review notes? Or are you ready for another review?

**Comment by JamesTessmer** [comment]
@mslarae13 yup, I didn’t get too much time to work on them before going on vacation, but I’m working on that today.

**Comment by pkalita-lbl** [comment]
The currently open PR (https://github.com/microbiomedata/nmdc-server/pull/1208) gets the main functionality for creating GitHub issues in place. The two caveats are:

- It is using a GitHub Personal Access Token to authorize the GitHub API request. This isn't ideal long-term, because we don't want to have these issues tied to James' account.
- It does not add the new issue to a project board. The authorization that's being used to create the issue can't be used to do this step.

The good news is that using a GitHub App for authorization solves both of those issues. I've created two issues outlining that work:

- https://github.com/microbiomedata/issues/issues/693
- https://github.com/microbiomedata/issues/issues/694

**Comment by mslarae13** [comment]
Thanks @pkalita-lbl ! I reviewed the PR on #1208 & the issue that gets created looks good!

Can we close this issue, and allow the 2 you made to track these improvements?

**Comment by ssarrafan** [comment]
Closing based on comments above from @mslarae13 and @pkalita-lbl
